i6_models/assemblies/e_branchformer/e_branchformer_v1.py

```diff
@@ -0,0 +1,124 @@
+    """
+    Ebranchformer block module
+    """
+
+    def __init__(self, cfg: EbranchformerBlockV1Config):
+        """
+        :param cfg: e_branchformer block configuration with subunits for the different e_branchformer parts
+        """
+        super().__init__()
+        self.ff_1 = PositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.mhsa = MHSAV1(cfg=cfg.mhsa_cfg)
+        self.cgmlp = ConvolutionalGatingMLPV1(model_cfg=cfg.cgmlp_cfg)
+        self.merger = MergerV1(model_cfg=cfg.merger_cfg)
+        self.ff_2 = PositionwiseFeedForwardV1(cfg=cfg.ff_cfg)
+        self.final_layer_norm = torch.nn.LayerNorm(cfg.ff_cfg.input_dim)
+
+    def forward(self, x: torch.Tensor, /, sequence_mask: torch.Tensor) -> torch.Tensor:
+        """
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: mask tensor where 0 defines positions within the sequence and 1 outside, shape: [B, T]
+        :return: torch.Tensor of shape [B, T, F]
+        """
+        x = 0.5 * self.ff1(x) + x  #  [B, T, F]
+        x_1 = self.mhsa(x, sequence_mask)  #  [B, T, F]
+        x_2 = self.cgmlp(x)  #  [B, T, F]
+        x = self.merger(x_1, x_2) + x  #  [B, T, F]
+        x = 0.5 * self.ff2(x) + x  #  [B, T, F]
+        x = self.final_layer_norm(x)  # [B, T, F]
+        return x
+
+
+class EbranchformerEncoderV1Config(ModelConfiguration):
+    """
+    Attributes:
+        num_layers: Number of e-branchformer layers in the e-branchformer encoder
+        frontend: A pair of ConformerFrontend and corresponding config
+        block_cfg: Configuration for EbranchformerBlockV1
+    """
+
+    num_layers: int
+
+    # nested configurations
+    frontend: ModuleFactoryV1
+    block_cfg: EbranchformerBlockV1Config
+
+
+class EbranchformerrEncoderV1(nn.Module):
+    """
+    Implementation of the Branchformer with Enhanced merging (short E-branchformer), as in the original publication.
+    The model consists of a frontend and a stack of N e-branchformer blocks.
+    C.f. https://arxiv.org/pdf/2210.00077.pdf
+    """
+
+    def __init__(self, cfg: EbranchformerEncoderV1Config):
+        """
+        :param cfg: conformer encoder configuration with subunits for frontend and conformer blocks
+        """
+        super().__init__()
+
+        self.frontend = cfg.frontend()
+        self.module_list = torch.nn.ModuleList([EbranchformerBlockV1(cfg.block_cfg) for _ in range(cfg.num_layers)])
+
+    def forward(self, data_tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        :param data_tensor: input tensor of shape [B, T', F]
+        :param sequence_mask: mask tensor where 0 defines positions within the sequence and 1 outside, shape: [B, T']
+        :return: (output, out_seq_mask)
+            where output is torch.Tensor of shape [B, T, F'],
+            out_seq_mask is a torch.Tensor of shape [B, T]
+
+        F: input feature dim, F': internal and output feature dim
```

Comment:
for time you use `T'` for the input / unmodified time (which makes sense imo) but I feel like then here `F` and `F'` should also be flipped.